src/index.ts

```diff
@@ -4,57 +4,69 @@ import { validateStates } from "./validate.js";
 const PREVENT_COSTRUCTOR_INSTANCE = Symbol("prevent-constructor");
 
 export type StateIdentifier = string;
-export type TransitionToHook = (
-	hook: HookInput,
+export type TransitionToHook<TContext = void, TEvent = void> = (
```

Comment:
```suggestion
export type TransitionToHook<TContext = unknown, TEvent = unknown> = (
```

I would apply the same for the others as well